### ISSUE TITLE
Since 10.18, copy between two file shares create top level dir

### ISSUE THREAD

**Issue by Aohzan** [Azure/azure-storage-azcopy#2141]
### Which version of the AzCopy was used? 
10.18.0

### Which platform are you using? (ex: Windows, Mac, Linux)
Linux

### What command did you run?
```bash
azcopy copy \
                --overwrite=ifSourceNewer \
                --recursive=true \
                --preserve-smb-info=true \
                --preserve-permissions=true \
                --exclude-path=tmp \
                "https://originstorageaccount.file.core.windows.net/myfileshare/?token" \
                "https://deststorageaccount.file.core.windows.net/myfileshare/?token"
```

I tried with remove trailing slashes but same results

### What problem was encountered?
the origin fileshare has something like
```
root
  |- subdir
     |- file1
     |- file2
```
after 10.18 upgrade, copy leads to
```
root
  |- myfileshare
      |- subdir
         |- file1
         |- file2
```

### How can we reproduce the problem in the simplest way?
run the copy command and see if a top level directory with the share name is created

### Have you found a mitigation/solution?
Downgrading to 10.17

**Comment by siminsavani-msft** [comment]
Hi @Aohzan ! Since you are attempting to preserve permissions with this flag `--preserve-permissions=true`, AzCopy won't strip the top directory, which creates the top level 'myfileshare' directory as you indicated.

**Comment by Aohzan** [comment]
Hello @siminsavani-msft, `--as-subdir=false` fix the problem indeed, thank you

**Comment by adreed-msft** [comment]
> Hello @siminsavani-msft, `--as-subdir=false` fix the problem indeed, thank you

Great to hear it could temporarily mitigate the issue at hand. I'll be reproducing & fixing this for release in 10.19.0 shortly.

**Comment by adreed-msft** [comment]
This has been fixed in AzCopy v10.18.1
